src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/HashPartitioningConditionMapper.java

```diff
@@ -0,0 +1,367 @@
+            request.putExpressionAttributeValue(placeholder, value);
+            return placeholder;
+        }
+
+        private String setFieldPlaceholder(RequestWrapper request, String field) {
+            String placeholder = MappingUtils.getNextFieldPlaceholder(request);
+            request.putExpressionAttributeName(placeholder, field);
+            return placeholder;
+        }
+    }
+
+    private static class BetweenRange {
+
+        private final AttributeValue lower;
+        private final AttributeValue upper;
+
+        BetweenRange(AttributeValue lower, AttributeValue upper) {
+            this.lower = lower;
+            this.upper = upper;
+        }
+    }
+
+    private static class KeyConditionExpressionVisitor extends ExpressionsBaseVisitor<Void> {
+
+        private final PrimaryKey primaryKey;
+        private final Map<String, String> fieldPlaceholders;
+        private final Map<String, AttributeValue> valuePlaceholders;
+
+        // output
+        private AttributeValue hkValue = null;
+        private Optional<KeyFieldCondition> rkCondition = Optional.empty();
+
+        KeyConditionExpressionVisitor(PrimaryKey primaryKey, Map<String, String> fieldPlaceholders,
+                                      Map<String, AttributeValue> valuePlaceholders) {
+            this.primaryKey = primaryKey;
+            this.fieldPlaceholders = fieldPlaceholders;
+            this.valuePlaceholders = valuePlaceholders;
+        }
+
+        @Override
+        public Void visitKeyConditionPart(KeyConditionPartContext keyConditionPart) {
+            IdContext id = keyConditionPart.id();
+            if (id != null) {
+                String fieldName = keyConditionPart.id().getText();
+                if (fieldName.startsWith("#")) {
+                    // if it's a field placeholder, remove it -- we'll generated a new one for translated expression
+                    fieldName = removeFieldPlaceholder(fieldName);
+                }
+
+                if (primaryKey.getHashKey().equals(fieldName)) {
+                    checkArgument(hkValue == null, "Multiple hash key conditions");
+                    ComparatorContext comparator = keyConditionPart.comparator();
+                    checkArgument(comparator != null && comparator.getText().equals("="),
+                        "Hash key condition must have operator '='");
+                    String valuePlaceholder = keyConditionPart.literal(0).getText();
+                    hkValue = removeValuePlaceholder(valuePlaceholder);
+                } else if (primaryKey.getRangeKey().isPresent() && primaryKey.getRangeKey().get().equals(fieldName)) {
+                    checkArgument(rkCondition.isEmpty(), "Multiple range key conditions");
+                    ComparatorContext comparator = keyConditionPart.comparator();
+                    ComparisonOperator op = comparator != null ? getComparatorType(comparator.getText()) : BETWEEN;
```

Comment:
Why does it default to `BETWEEN`?

Reply:
Oh was thinking of random partitioning ... makes sense that it defaults to between.

Reply:
This is because in the ANTLR grammar definition, we have a rule called `comparator` that defines the single-value comparators (i.e., `=`, `<>`, `>`, `>=`, `<`, `<=`), whereas `BETWEEN` is defined as a separate lexer rule, since its syntax is different (`id comparator literal` vs. `id BETWEEN literal AND literal`). ComparatorContext here corresponds to the `comparator` rule. And based on the way the `keyConditionExpression` rule is defined, when there is an `id`, and there is no `comparator`, there should be a `BETWEEN` token. But maybe it'd be better if I check that explicitly, instead of defaulting to BETWEEN like this.